src/kubernetes_cluster/spec/kubernetes_api/state_machine.rs

```diff
@@ -176,6 +180,8 @@ pub open spec fn validate_update_request(req: UpdateRequest, s: KubernetesAPISta
         && req.obj.metadata.resource_version != s.resources[req.key].metadata.resource_version {
         // Update fails because the object has a wrong rv
         Option::Some(APIError::Conflict)
+    } else if req.obj.kind == Kind::CustomResourceKind && !K::transition_rule(req.obj, s.resources[req.key]) {
```

Comment:
Don't we need to apply `rule` for update request as well? I think k8s will check `rule` for both create and update.